File: Data Structures & Algorithms/majority-element/submission-0.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;

        for (int i = 0 ; i < n ; i++){
            int sum = 1;

            for (int j = i + 1 ; j < n ; j++){
                if (nums[j] == nums[i]) sum++;
            }

            if (sum > (n / 2)) return nums[i];
        }

        return 0;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxProduct(nums) {
        let res = nums[0];

        for (let i = 0 ; i < nums.length ; i++){
            let curr = nums[i];

            res = Math.max(curr, res);

            for (let j = i+1 ; j < nums.length; j++ ){
                curr *= nums[j];

                res = Math.max(curr, res);
            }
        }

        return res;
    }
}
